### PR TITLE
Don't skip blocks passed end and simplify the Seek implementation.

### PR DESCRIPTION
**Pull request by ahsonkhan**
Borrows the tests from https://github.com/dotnet/corefx/pull/28261 by @benaadams and builds on top of it.

- Ignore the formatting/clean up changes.
- The most important change is to the Seek and SeekMultiSegment methods. With this simplification, the fix in https://github.com/dotnet/corefx/pull/28261 is no longer necessary, since this PR supersedes it.
   - This change greatly simplifies the implementation (and the disassembly shrinks, i.e. this will almost certainly help performance). The loop body shrinks to 14 instructions, from 23.

**Before (loop highlighted):**
image

**After (loop highlighted):**
image

cc @AlexRadch, @pakrym, @davidfowl, @joshfree, @benaadams

### CODE REVIEW

**Comment by ahsonkhan** [comment]
@dotnet-bot test OSX x64 Debug Build

OSX.1013.Amd64.Open-x64:Debug
https://mc.dot.net/#/user/ahsonkhan/pr~2Fjenkins~2Fdotnet~2Fcorefx~2Fmaster~2F/test~2Ffunctional~2Fcli~2F/026ad131c40eea4c339dcf942731ebba6bfbc3f2/workItem/System.Net.Http.Functional.Tests/wilogs

**Comment by pakrym** [comment]
> Ignore the formatting/clean up changes.

It's easier to review when style changes are in the separate commit.

File: src/System.Memory/src/System/Buffers/ReadOnlySequence_helpers.cs [hunk 1]
```diff
@@ -221,66 +187,52 @@ internal SequencePosition Seek(in SequencePosition start, in SequencePosition en
...
-                        break;
-                    }
-
-                    // Move to next
-                    count -= memoryLength;
-                    currentSegment = currentSegment.Next;
-                }
+                int memoryLength = currentSegment.Memory.Length;
+
+                // Fully contained in this segment; or at end of segment but there isn't a next one to move to
```
Comment (pakrym) [inline src/System.Memory/src/System/Buffers/ReadOnlySequence_helpers.cs]: This comment is not correct anymore without `memoryLength == count && isCurrentAtEnd` this check/

**Comment by ahsonkhan** [comment]
@dotnet-bot test OSX x64 Debug Build

https://github.com/dotnet/corefx/issues/28342

**Comment by ahsonkhan** [comment]
@dotnet-bot test Linux x64 Release Build
@dotnet-bot test Windows x86 Release Build

**Pull request merged**